src/services/pfs/lib/FileClient.ts

```diff
@@ -32,7 +32,8 @@ export class FileClient<T> {
   }
 
   putFileFromBytes(path: string, bytes: Buffer, callback?: () => void) {
-    const messageLength = GRPC_MAX_MESSAGE_LENGTH - STREAM_OVERHEAD_LENGTH;
+    const messageLength =
+      GRPC_MAX_MESSAGE_LENGTH - STREAM_OVERHEAD_LENGTH - path.length;
```

Comment:
I'm glad we talked about this, it would have been a weird bug to track down.

Reply:
Yeah good catch on this one.